homeassistant/components/freebox/alarm_control_panel.py

```diff
@@ -110,16 +110,12 @@ def _set_features(self, node: dict[str, Any]) -> None:
         can_arm_home = False
         for nodeid, local_node in self._router.home_devices.items():
             if nodeid == local_node["id"]:
-                alarm2 = next(
-                    filter(
-                        lambda x: (x["name"] == "alarm2" and x["ep_type"] == "signal"),
-                        local_node["show_endpoints"],
-                    ),
-                    None,
-                )
-                if alarm2:
-                    can_arm_home = alarm2["value"]
-                    break
+                for endpoint in local_node["show_endpoints"]:
+                    if endpoint["name"] == "alarm2" and endpoint["ep_type"] == "signal":
+                        can_arm_home = endpoint["value"]
```

Comment:
We're no longer checking if the endpoint is truthy before accessing it. Is that not needed?

Reply:
No, if endpoint["name"] == "alarm2" and endpoint["ep_type"] == "signal" is true, endpoint["value"] is set

Reply:
Based on test fixtures, either:
- go check to `node["show_endpoints"]` -> `"name": "alarm2"` + `"ep_type": "slot"` (yes, **slot**) -> check `"value": true`
- go check `node["type"]["endpoints"]` (yes, **["type"]["endpoints"]**)  -> `"name": "alarm2"` + `"ep_type": "signal"` presence

Otherwise, it's not working

Reply:
` # Search if the arm home feature is present => has an "alarm2" endpoint
        can_arm_home = next(
            (
                endpoint
                for endpoint in self._node["type"]["endpoints"]
                if endpoint["name"] == "alarm2" and endpoint["ep_type"] == "slot"
            ),
            None,
        )

#if  search feature not None alarm2 is found

        if (
            self._command_arm_home and can_arm_home is not None
        ):  # or check self.get_value("slot", "alarm2") ???
            self._attr_supported_features = (
                AlarmControlPanelEntityFeature.ARM_AWAY
                | AlarmControlPanelEntityFeature.ARM_HOME
                | AlarmControlPanelEntityFeature.TRIGGER
            )`